test/libsolidity/smtCheckerTests/imports/ExtCall.sol

```diff
@@ -0,0 +1,48 @@
+==== Source: ExtCall.sol ====
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.6;
+
+interface Unknown {
+	function callme() external;
+}
+
+contract ExtCall {
+	uint x;
+
+	bool lock;
+	modifier mutex {
+		require(!lock);
+		lock = true;
+		_;
+		lock = false;
+	}
+
+	function setX(uint y) mutex public {
+		x = y;
+	}
+
+	function xMut(Unknown u) public {
+		uint x_prev = x;
+		u.callme();
+		assert(x_prev == x);
+	}
+}
+==== Source: ExtCall.t.sol ====
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.6;
+
+import "ExtCall.sol";
+
+contract ExtCallTest {
+    ExtCall call;
+
+    function setUp() public {
+        call = new ExtCall();
+    }
+}
+// ====
+// SMTEngine: all
+// SMTIgnoreCex: yes
+// ----
+// Warning 6328: (ExtCall.sol:360-379): CHC: Assertion violation happens here.
+// Warning 4588: (ExtCall.t.sol:180-193): Assertion checker does not yet implement this type of function call.
```

Comment:
This does not produce an ICE for me. Checked on both `develop` and 0.8.10 release binary. I remember a similar situation in https://github.com/ethereum/solidity/pull/11852#discussion_r697551335 but there I forgot to enable the SMTChecker. Here I tried with:
```bash
solc ExtCall.t.sol --model-checker-engine all
```
```
Warning: CHC: Assertion violation happens here.
Counterexample:
x = 1, lock = false
u = 0
x_prev = 0

Transaction trace:
ExtCall.constructor()
State: x = 0, lock = false
ExtCall.xMut(0)
    u.callme() -- untrusted external call, synthesized as:
        ExtCall.setX(1) -- reentrant call
  --> /tmp/ExtCall.t.sol:26:3:
   |
26 |            assert(x_prev == x);
   |            ^^^^^^^^^^^^^^^^^^^
```